### ISSUE TITLE
Output broken by git operations with ssh urls

### ISSUE THREAD

**Issue by glen-nicol** [adamralph/bullseye#404]
## Version(s)
3.2.0

## To reproduce

```csharp
Target("test", () =>
{
    // should fail to resolve but still print output without garbage
    Run("git", $"clone ssh://git@github.com:adamralph/bullseye.git tmp_bullseye"); 
});
```

## Expected behavior

The output should not contain garbage characters

## Actual behaviour

outputs the following
```
make: Starting... (test)
make: test: Starting...
make: git clone ssh://git@github.com:adamralph/bullseye.git CloneTmp
Cloning into 'CloneTmp'...
ssh: Could not resolve hostname github.com:adamralph: Name or service not known
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
?[90mmake:?[0m ?[36mtest?[37m:?[0m ?[91mFailed! The command exited with code 128.?[0m ?[35m(142 ms)?[0m?[0m
?[90mmake:?[0m ?[37m-------------------------------?[0m
?[90mmake:?[0m ?[37mDuration?[0m        ?[37mOutcome?[0m  ?[37mTarget?[0m
?[90mmake:?[0m ?[37m--------------?[0m  ?[37m-------?[0m  ?[37m------?[0m
?[90mmake:?[0m ?[35m142 ms?[0m?[0m  ?[35m100.0%?[0m?[0m  ?[91mFailed!?[0m?[0m  ?[36mtest?[0m  ?[0m
?[90mmake:?[0m ?[37m-------------------------------?[0m
?[90mmake:?[0m ?[91mFailed!?[0m ?[36m(test)?[0m ?[35m(148 ms)?[0m?[0m
```

## Workarounds

It appears HTTPS urls do not cause this to happen. But that is not viable for our internal git server which requires SSH. 

## Additional context

This occurs with clone, and push operations that involve an SSH url. Not sure what git is doing or outputting for them. I also tried the new -E option from #393 with no luck

**Comment by adamralph** [comment]
The garbage characters appear because the environment you are running in does not support ANSI colouring. That is easy to fix. Just use the `-N`/`--no-color` option when invoking Bullseye.

The SSH failure is unrelated to your use of Bullseye (or SimpleExec). If you want to clone from an SSH URL you must configure Git to use appropriate credentials.

**Comment by glen-nicol** [comment]
I am no expert on ansi coloring, but color seems to work on everything in my terminal. If I use an https url in the code above it will print the bullseye report in color just fine. The SSH failure isn't important it was just the easiest way to repro the problem without using a private URL that clones just fine internally.

**Comment by adamralph** [comment]
OK, I see what you mean:

![image](https://user-images.githubusercontent.com/677704/75271909-48539100-57fd-11ea-87ef-b6db19c45229.png)

It seems to be something related to running git specifically. And it doesn't matter whether the operation fails or succeeds:

![image](https://user-images.githubusercontent.com/677704/75272000-6b7e4080-57fd-11ea-96b0-0058e59fa876.png)

I'm not yet sure where the problem lies but I'll try and investigate it. It could be something in SimpleExec rather than Bullseye, or it may be something wonky that git is doing to the terminal, in which case the fix will lie over there.

**Comment by adamralph** [comment]
OK, exactly what I thought is happening, is happening. Git is tampering with the console mode. Before Git executes, the console mode has the `ENABLE_VIRTUAL_TERMINAL_PROCESSING` (`0x0004`) flag set, and after Git executes, that flag is no longer set.

This was [mentioned over 2 years ago](https://github.com/git-for-windows/git/issues/1121#issuecomment-323047927) in the Git for Windows issue tracker.

So in summary, this is a Git for Windows problem and needs to be pursued over there. In the meantime, the workaround is to run Bullseye with the `-N`/`--no-color` option.

Thanks for raising this!

**Comment by glen-nicol** [comment]
Huh, I am not sure I would have been able to track that down. Thanks for looking into it for me! I will see what we can do that git issue.
